node.go

```diff
@@ -55,6 +64,7 @@ type MerkleNode interface {
 }
 
 type ProofNode interface {
```

Comment:
Users need to recompute the hash of the proof node, so they should need access to all the data included in the respective leaf hashes such as the ```commitment```, the ```treeNonce``` etc.